### ISSUE TITLE
Exporting CSV Error

### ISSUE THREAD

**Issue by eliu-ual** [digital-preservation/droid#1130]
I have a couple of .droid files I’m trying to export to CSV, but I’m receiving an error about an illegal character preventing DROID from generating a CSV file. I’ve attached the beginning of the error message to this email (the full error is very long, but the remainder is just the rest of the filepath).

For background, I’m running DROID on a set of files to support identification of duplicate records. The hard drive was as donated Mac-formatted hard drive, then a copy was made onto a Windows network drive. I’m running DROID on the Windows copy, and I’m aware there can be loads of issues going from Mac to Windows as they differ in tolerance to ‘non-standard’ character usage such as backslashes etc.

Is this a common error? Might it be resolved by running a Mac-compatible version of DROID on a Mac-formatted copy?

Thanks in advance for any advice you might have!
![Screenshot 2024-08-15 174823](https://github.com/user-attachments/assets/90e447ce-2b64-44e0-b822-d73a467100f9)

**Comment by steve-daly** [comment]
DROID is cross-platform so it's the same code running on every operating system, but I can see that underlying file system behaviour can vary.   Can you see what the folder/file names look like in Windows?  DROID should be fine with unicode characters in filenames but obviously something unusual is happening here.  If you can see an unusual filename could you possibly copy that filename onto a harmless file (e.g. blank text document) and upload the document here to see if we can reproduce this with DROID.

**Comment by eliu-ual** [comment]
Hi Steve! Thanks for your prompt response. Strangely the filename cited in the error message is pretty simple, just 'Icon'. But I can see some of the other filenames in the same folder could be problematic; these are 'C=P_Ars_electronica_edit3.doc' and 'Ars-final.doc'.

I've attached .txt files with each of these names to support testing.

[Icon.txt](https://github.com/user-attachments/files/17211008/Icon.txt)
[Ars-final.doc.txt](https://github.com/user-attachments/files/17211074/Ars-final.doc.txt)
[C=P_Ars_electronica_edit3.doc.txt](https://github.com/user-attachments/files/17211080/C.P_Ars_electronica_edit3.doc.txt)

**Comment by steve-daly** [comment]
Thanks.  In your original screenshot you have some scissors emojis in the filepath. Do you see that in your folder browser/explorer too.

**Comment by eliu-ual** [comment]
In File Explorer, they appear as boxes - definitely a legacy of material being moved from Mac to Windows: 

S:\DAC Projects\Archives and Special Collections Centre\Her Noise Archive\2_HNPLANNING\2_HN_PLANNING\HN12\1_ORGANISERS\HN123\3_Lina\HN12310\PREVIOUS WRITING\Ars Electronica 02 (The Wire)

**Comment by aladrick** [comment]
I am having a similar issue but the file name and the path don't have any special characters in them. There might be a space at the end of the filename but I've been unable to change it at all so I'm not sure what to do.

**Comment by steve-daly** [comment]
@MancunianSam if you have a moment, can you see how DROID behaves with trailing spaces on filenames?

**Comment by sparkhi** [comment]
I believe the trailing space (typed using the space bar) is fine. I suspect it might be some other non-printable character at the end of that filename rather than a "space"

**Comment by MancunianSam** [comment]
I've managed to recreate this by exporting a profile containing a file with a space at the end from a Linux machine and opening it on Windows and trying to export. I get the error in this issue. 

I've raised a linked pull request which trims the space at the end before export if DROID is running on Windows and this has fixed the issue for me. I'm going to try to get a interim release out soon so you can try this out.

**Comment by MancunianSam** [comment]
Good point. The error is in converting the url file:///C/path/to/file%20 into C:\path\to\file because you can't have a space on the end. So I've changed it to remove the space to allow the conversion and then put the space back so it exports correctly.

**Comment by ross-spencer** [comment]
Sure thing! 

And just done -- 

![Image](https://github.com/user-attachments/assets/3d874ea8-ed0a-4b7d-a371-8b916d9252b3)

LGTM! thx Sam!

**Comment by aladrick** [comment]
Thanks so much for your work on this, I really appreciate it! Unfortunately I'm still having the same problem but I think it's something to do with the folder itself—I can't edit or delete it because it says the folder does not exist. Thanks again, though!
